modules/servers/upCloudVps/lib/upCloudVps.php

```diff
@@ -234,6 +235,35 @@ public function CreateServer($ZoneID, $Hostname, $Plan, $OsUUID, $sshKey, $user_
   ];
 }
 
+if ($TemplateType == 'native') {
+    if (!preg_match('/Windows/', $TemplateTitle)) {
+        if ($sshKey != "na") {

```

Comment:
Curiosity: does this `"na"` come from the frontend as the default form value?

Reply:
Not quite, but it gets persisted as a placeholder in db on create, https://github.com/UpCloudLtd/upcloud-whmcs-module/blob/f57d1b54d09587e46fe16af7c267d4743fce1cd3/modules/servers/upCloudVps/lib/vmManager.php#L66-L67